1_Get_Started/practice/prototypes_gim.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+function randMax(max) {
+  return Math.trunc(1e9 * Math.random()) % max;
+}
+
+function getSymbolRangeOf(idx, symbols) {
+  if (idx < 0) {
+    return symbols[symbols.length - 1];
+  }
+  return symbols[idx % symbols.length];
+}
+
+var reel = {
+  symbols: ["♠", "♥", "♦", "♣", "☺", "★", "☾", "☀"],
+  spin() {
+    if (this.position == null) {
+      this.position = randMax(this.symbols.length - 1);
+    }
+    this.position = (this.position + 100 + randMax(100)) % this.symbols.length;
+  },
+  display() {
+    if (this.position == null) {
+      this.position = randMax(this.symbols.length - 1);
+    }
+    return this.symbols[this.position];
+  },
+};
+
+var slotMachine = {
+  reels: [Object.create(reel), Object.create(reel), Object.create(reel)],
+  spin() {
+    this.reels.forEach(function spinReel(reel) {
+      reel.spin();
+    });
+  },
+  display() {
+    const moves = [-1, 0, 1];
+    const reelRows = [];
+    for (const move of moves) {
+      const reelRow = [];
+      this.reels.forEach(reel => {
+        const curPosition = reel.position + move;
+        const symbol = getSymbolRangeOf(curPosition, reel.symbols);
+        reelRow.push(symbol);
```

Comment:
물론 스타일 차이겠지만, 짧은 코드에 오히려 변수를 생성함으로인해 복잡도가 늘어난것 같아서 저는 아래와 같이 이 부분은 합치는게 더 깔끔해 보여서 리뷰했습니다. 
 
```suggestion
        reelRow.push(
            getSymbolRangeOf(reel.position + move, reel.symbols)
        );
```